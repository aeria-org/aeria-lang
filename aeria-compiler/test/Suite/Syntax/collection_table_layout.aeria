collection CollectionTableLayoutTest {
  tableLayout {
    action1 {
      route "/route1"
      icon "fa fa-icon"
      ask true
      if @cond(prop1 == true)
      selection false
      effect "foo"
      button true
      translate true
      setItem true
      fetchItem true
      clearItem true
      params @js () => ({
        prop1: "value1",
        prop2: "value2"
      })
      @end
      query @js () => ({
        prop3: "value3",
        prop4: "value4"
      })
      @end
      requires {
        prop1
        prop2
        prop3
      }
    }

    action2 {
      button false
    }
  }
}
