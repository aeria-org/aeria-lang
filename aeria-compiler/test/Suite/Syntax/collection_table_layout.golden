{
  "kind": "Program",
  "collections": [
    {
      "tableLayout": [
        {
          "route": "/route1",
          "kind": "TableLayoutItem",
          "if_": {
            "kind": "Cond",
            "expr": {
              "right": {
                "kind": "ELiteral",
                "expr": {
                  "value": true,
                  "kind": "LBoolean"
                }
              },
              "left": {
                "kind": "ELiteral",
                "expr": {
                  "value": {
                    "value": "prop1",
                    "kind": "PropertyName"
                  },
                  "kind": "LProperty"
                }
              },
              "kind": "EEq"
            }
          },
          "button": {
            "value": true,
            "type": "left"
          },
          "actionName": {
            "value": "action1",
            "kind": "PropertyName"
          },
          "action": {
            "translate": true,
            "setItem": true,
            "selection": false,
            "requires": [
              {
                "value": "prop1",
                "kind": "PropertyName"
              },
              {
                "value": "prop2",
                "kind": "PropertyName"
              },
              {
                "value": "prop3",
                "kind": "PropertyName"
              }
            ],
            "query": {
              "macro": "({\n        prop3: \"value3\",\n        prop4: \"value4\"\n      })\n      ",
              "kind": "Macro"
            },
            "params": {
              "macro": "({\n        prop1: \"value1\",\n        prop2: \"value2\"\n      })\n      ",
              "kind": "Macro"
            },
            "kind": "ActionItem",
            "icon": "fa fa-icon",
            "fetchItem": true,
            "effect": "foo",
            "clearItem": true,
            "ask": true,
            "actionName": {
              "value": "action1",
              "kind": "PropertyName"
            }
          }
        },
        {
          "kind": "TableLayoutItem",
          "button": {
            "value": false,
            "type": "left"
          },
          "actionName": {
            "value": "action2",
            "kind": "PropertyName"
          },
          "action": {
            "requires": [],
            "kind": "ActionItem",
            "actionName": {
              "value": "action2",
              "kind": "PropertyName"
            }
          }
        }
      ],
      "name": {
        "value": "CollectionTableLayoutTest",
        "kind": "CollectionName"
      },
      "kind": "Collection"
    }
  ]
}
